bin/pcl/src/auth.rs

```diff
@@ -0,0 +1,165 @@
+@@@@@   @@@@@@@@@@  @@@q___  ___g@@B  @@@@@@@@@   .@@@@@
+@@@@@@   @@@@@@@@@  [@@@@@@  @@@@@@|  @@@@@@@@P   g@@@@@
+@@@@@@\  '@@@@@@@@,  @@@@@@  @@@@@@  |@@@@@@@W   /@@@@@@
+@@@@@@@L  `@@@@@@@@  0@@@@g  @@@@@F  @@@@@@@P   /@@@@@@@
+@@@@@@@@p   \@@@@@@,  @@@@8  @@@@W  A@@@@@B    j@@@@@@@@
+@@@@@@@@@@_   "@@@@@,  @@@]  @@@D  /@@@@D    _@@@@@@@@@@
+@@@@@@@@@@@@_    <B@@_  <=   "8"  /@BP"    _@@@@@@@@@@@@
+@@@@@@@@@@@@@@@_     ""          "      _g@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@g__              __g@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+"#;
+
+#[derive(Serialize, Deserialize)]
+struct AuthResponse {
+    code: String,
+    sessionId: String,
+    deviceSecret: String,
+    expiresAt: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct StatusResponse {
+    verified: bool,
+    address: Option<String>,
+    token: Option<String>,
+    refresh_token: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct StoredAuth {
+    access_token: String,
+    refresh_token: String,
+    address: String,
+}
+
+pub async fn login() -> Result<()> {
+    let client = Client::new();
+
+    // Get authentication code
+    let auth_response: AuthResponse = client
+        .get(format!("{}/api/v1/cli/auth/code", BASE_URL))
+        .send()
+        .await?
+        .json()
+        .await?;
+
+    // Display code and instructions to user
+    println!("\n🔑 Your authentication code is:\n");
+    println!("    ┌──────────┐");
+    println!("    │  {}  │", auth_response.code);
+    println!("    └──────────┘\n");
+    println!("📱 Please visit:");
+    println!(
+        "   {}/device?session_id={}\n",
+        BASE_URL, auth_response.sessionId
+    );
+    println!("⏳ Waiting for authentication...\n");
+
+    // Poll for authentication status
+    let mut attempts = 0;
+    while attempts < MAX_RETRIES {
+        let status: StatusResponse = client
+            .get(format!("{}/api/v1/cli/auth/status", BASE_URL))
+            .query(&[
+                ("session_id", &auth_response.sessionId),
+                ("device_secret", &auth_response.deviceSecret),
+            ])
+            .send()
+            .await?
+            .json()
+            .await?;
+
+        if status.verified {
+            // Store authentication tokens
+            let auth = StoredAuth {
+                access_token: status.token.unwrap(),
+                refresh_token: status.refresh_token.unwrap(),
+                address: status.address.unwrap(),
+            };
+            store_auth(&auth)?;
+            println!("{}", PHYLAX_ASCII);
+            println!("\n🎉 Authentication successful!");
+            println!("🔗 Connected wallet: {}\n", auth.address);
+            return Ok(());
+        }
+
+        attempts += 1;
+        sleep(POLL_INTERVAL).await;
+    }
+
+    Err(eyre::eyre!("Authentication timed out"))
+}
+
+pub fn logout() -> Result<()> {
```

Comment:
ideally, prints should stay in the bin level (we have to change that) so basically you consume whatever functionality your crate offers (in this case the logout()) and the nyou print according to the results